src/main/java/ui/components/Notification.java

```diff
@@ -0,0 +1,60 @@
+package ui.components;
+
+import javafx.scene.Node;
+
+import java.util.Optional;
+
+public class Notification {
+
+    public static final int DEFAULT_NOTIFICATION_PANE_VISIBLE_PERIOD = 10;
+
+    public enum NotificationType { NOACTION, ACTIONONBUTTON, ACTIONONTIMEOUT, ACTIONONBUTTONANDTIMEOUT }
+
+    private final NotificationType notificationType;
+    private final Node icon;
+    private final String message;
+    private final Optional<String> buttonLabel;
+    private final Optional<Runnable> timeoutRunnable;
+    private final Optional<Runnable> buttonRunnable;
```

Comment:
These don't have to be optional. Runnables have an 'empty' value: the no-op `() -> {}`. We could default to no-ops when runnables are absent. The interface could then be slightly higher-level, with `timeout` and `action` (for example) methods which just `run` the corresponding actions.

Is it possible to have a button without a button label? If not, it probably doesn't have to be optional also.